packages/grafana-ui/src/components/Segment/Segment.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { useState } from 'react';
+import { SegmentSelect } from './SegmentSelect';
+import { OptionType } from './GroupBy';
+
+export interface Props {
+  value?: string;
+  options: OptionType;
```

Comment:
think options should be an async callback so we can have a lazy dropdown. the value should be T (generic) to support number and string values

Maybe we should use SelectableValue<T> here for both value and getOptions.

Reply:
Yes!